provider/azureProvider.js

```diff
@@ -112,16 +92,14 @@ return this.parsedBindings;
     }
   }
 
-  LoginWithServicePrincipal () {
-return new BbPromise((resolve, reject) => {
-      msRestAzure.loginWithServicePrincipalSecret(servicePrincipalClientId, servicePrincipalPassword, servicePrincipalTenantId, (error, credentials) => {
-        if (error) {
-          reject(error);
-        } else {
-          principalCredentials = credentials;
-          resolve(credentials);
-        }
-      });
+  Login() {
+    return login({ interactiveLoginHandler: (code) => {
+        this.serverless.cli.log(`Paste this code (copied to your clipboard) into the launched browser, and complete the authentication process: ${code}`);
```

Comment:
I agree that new Azure customers wouldn't be familiar with SPs, which is why this PR doesn't currently log out that it's creating one. The auto-generated SP is meant to be an implementation detail that simply provides the ability to login interactively once, and then continue to use the CLI without needing to login again. I'd personally vote not to add any logging here, because it might just complicate the experience.